### PR TITLE
fix(regex): [BACK-1457]: Fix regex to parse de collections correctly

### PR DESCRIPTION
**Pull request by Herraj**
## Goal
When manually adding a **collection** url in the tool, the `isCollection` flag was not being set to `true` for `de` collections. The front-end was working as expected so the problem seemed to be in the regex that was parsing the collection url. I didn't account for an optional `.../de/...` in the url for de collections. 

I have also made the function flexible where if in the future we decide to support more collection languages, we would only need to update the `CollectionLanguage` enum.

Tickets:

- [BACK-1457]

[BACK-1457]: https://getpocket.atlassian.net/browse/BACK-1457?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/public/resolvers/item.ts [hunk 1]
```diff
@@ -12,9 +12,8 @@ export async function collection(
   { dataLoaders }
 ): Promise<Collection> {
   // match only http(s)://getpocket.com/collections/<slug>
-  const matches = /^https?:\/\/(?:getpocket\.com)\/collections\/(.*)/i.exec(
-    givenUrl
-  );
+  const matches =
+    /^https?:\/\/(?:getpocket\.com)?(\/de)?\/collections\/(.*)/i.exec(givenUrl);
```
[the remark this answers is not in the document]
Comment (Herraj) [reply src/public/resolvers/item.ts:16]: Yep, I couldn't find any tests I deployed it for now to see if CI picks up any test failures but seems like we don't have any. I'll add tests before putting it up for review!

**Review by kschelonka — APPROVED** [review]

File: src/test/helpers.ts [hunk 2]
```diff
@@ -342,3 +342,28 @@ export const getServerWithMockedHeaders = (headers: {
...
+
+/**
+ * Determines if a url is a valid collection url and returns its slug.
+ * Returns null for invalid collection urls
+ * @param url
+ * @returns A collection's slug or null
+ */
+export const getCollectionUrlSlug = (url: string): string | null => {
+  // get a string of allowed languages in this format "en|de|..."
+  const allowedLanguages = Object.values(CollectionLanguage).join('|');
```
Comment (jpetto) [inline src/test/helpers.ts:354]: this is really nice and extensible!

File: src/test/public-server.ts [hunk 3]
```diff
@@ -18,6 +19,9 @@ export const getServer = () => {
     schema: buildSubgraphSchema([{ typeDefs: typeDefsPublic, resolvers }]),
     context: {
       db: client(),
+      dataLoaders: {
+        collectionLoader,
+      },
```
Comment (jpetto) [inline src/test/public-server.ts:24]: how was this working previously without `dataLoaders` defined here?

**Review by jpetto — APPROVED** [review]
really nice work! i do have a question on the `dataLoaders` property being added though...

File: src/test/public-server.ts (hunk 3 above)
Comment (Herraj) [reply src/test/public-server.ts:24]: The other tests in this file were testing functions that never used a dataloader, just direct db queries!

**Pull request merged**